test/packages/missing_version/manifest.yml

```diff
@@ -0,0 +1,22 @@
+format_version: 1.0.0
+name: inconsistent_version
+title: "Missing Version"
+license: basic
+description: "This is a test package."
+type: integration
+version:
```

Comment:
I'm surprised this passes validation, given https://github.com/elastic/package-spec/blob/ade1f5216c0d1320c83269f533b6531ec48ff85b/versions/1/manifest.spec.yml#L281 and https://github.com/elastic/package-spec/blob/ade1f5216c0d1320c83269f533b6531ec48ff85b/versions/1/manifest.spec.yml#L103.

Reply:
Actually, I'm not sure this will even pass basic YAML validation. So I'm wondering why we need this extra check for the missing version.

Reply:
Actually it passed somehow and resulted with panic. I found this issue with help of CI (see: https://github.com/elastic/elastic-package/pull/354).

Reply:
I think it's because we are not short-circuiting in our validation logic. Maybe if there are errors here we should short circuit and fail fast?

https://github.com/elastic/package-spec/blob/ade1f5216c0d1320c83269f533b6531ec48ff85b/code/go/internal/validator/spec.go#L64

Reply:
I like this idea. I adjusted the PR.